docs/typescript.md

```diff
@@ -2,9 +2,77 @@
 title: "Typescript"
 ---
 
-Emotion includes TypeScript definitions for `styled` components and has type inferences for both html elements and React components.
+Emotion includes TypeScript definitions for `create-emotion`, `emotion`, `create-emotion-styled`, and `react-emotion` packages. These definition also could infer types for css property (with object syntax), and HTML/SVG tag name, and props types.
+
+## create-emotion
+
+``` tsx
+import createEmotion, { Emotion, EmotionOption } from 'create-emotion';
+
+const context = {};
+const options: EmotionOption = {
+  key: 'my-emotion',
+};
+const myEmotion: Emotion = createEmotion(context, options);
+
+const titleStyle = myEmotion.css({
+  boxSizing: 'border-box',
+  width: 300,
+  height: 200,
+});
+
+const subtitleStyle = myEmotion.css`
+  boxSizing: 'border-box';
+  width: 100;
+  height: 60;
+`;
+```
+
+Typescript checks css properties in object style syntax using csstype package, so following code will emit errors.
+
+``` tsx
+import createEmotion, { Emotion, EmotionOption } from 'create-emotion';
+
+const context = {};
+const options: EmotionOption = {
+  key: 'my-emotion',
+};
+const myEmotion: Emotion = createEmotion(context, options);
+
+const titleStyle = myEmotion.css({
+                                 ^ Argument of type 'boxSizing: 'bordre-box';' is not assignable [...]
+  boxSizing: 'bordre-box', // Oops, there's a typo!
+  width: 300,
+  height: 200,
+});
+
+const subtitleStyle = myEmotion.css`
+  boxSizing: 'border-box';
+  width: 100;
+  height: 60;
+`;
+```
+
+## emotion
+
+Basically same with `create-emotion`.
+
+``` tsx
+import { css } from 'emotion';
 
-### html elements
+const bodyStyle = css({
+  display: 'flex',
+  flowDirection: 'column-reverse',
+});
+```
+
+## create-emotion-styled
+
+Current typing for `create-emotion-styled` is only compatible with React, and will not work with Preact. For detail typing, see following `react-element` section.
```

Comment:
Could you change "Current typing for `create-emotion-styled` is" to "The current typings for `create-emotion-styled` are only"?

Reply:
OK. Thank you for your review!